apps/Gruntfile.js

```diff
@@ -404,6 +404,8 @@ module.exports = function (grunt) {
 
     pd: './src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx',
 
+    teacherApplication: './src/code-studio/pd/teacher_application/teacher_application.jsx',
```

Comment:
To get rid of the warnings about entry points, just change this line to:

```js
teacher_application: './src/sites/studio/pages/pd/teacher_application/teacher_application.js'
```

The idea is pretty much that instead of inlining javascript in our haml pages, we just add a script tag that points to a js file built by webpack. And since inlined script tags have a one-to-one correspondence with the haml file they get inlined into, we figured it would make sense to maintain that one-to-one correspondence when moving the otherwise inlined code into a standalone file.

In this particular case, your `teacher_application.jsx` file is acting like both a "js module" meant to be imported by other js modules (since it has a default export) _and_ an "entry point" which is _not_ meant to be imported by other js modules (since it attaches things to the global window object).

I think what you really want is two files:

1. **TeacherApplication.jsx** - this is a js module that lives in `src/code-studio/pd/teacher_application/` and exports the react component you created. It's modular, reusable, and unit testable. :+1:
2. **teacher_application.js** - this is an _entry point_ that lives in `src/sites/studio/pages/pd/teacher_application` and which is only ever referenced with a `<script>` tag in the `teacher_application.html.haml` file. All it does is setup the page by importing `TeacherApplication` and rendering it to some location in the DOM. It's not modular, not reusable, and is so short and simple that there is no point in writing unit tests for it.